src/traccuracy/_run_metrics.py

```diff
@@ -13,43 +11,28 @@
 def run_metrics(
     gt_data: "TrackingGraph",
     pred_data: "TrackingGraph",
-    matcher: "Type[Matcher]",
-    metrics: "List[Type[Metric]]",
-    matcher_kwargs: "Optional[Dict]" = None,
-    metrics_kwargs: "Optional[Dict]" = None,  # weights
+    matcher: "Matcher",
+    metrics: "List[Metric]",
 ) -> "Dict":
     """Compute given metrics on data using the given matcher.
 
-    An error will be thrown if the given matcher is not compatible with
-    all metrics in the given list. The returned result dictionary will
-    contain all metrics computed by the given Metric classes, as well as
-    general summary numbers e.g. false positive/false negative detection
-    and edge counts.
+    The returned result dictionary will contain all metrics computed by
+    the given Metric classes, as well as general summary numbers
+    e.g. false positive/false negative detection and edge counts.
 
     Args:
-        gt_data (TrackingData): ground truth graph and optionally segmentation
-        pred_data (TrackingData): predicted graph and optionally segmentation
-        matcher (traccuracy.matchers.Matched): matching class to use to create correspondence
-        metrics (List[Metric]): list of metrics to compute as class names
-        matcher_kwargs (optional, dict): Dictionary of keyword argument for the
-            matcher class
-        metric_kwargs (optional, dict): Dictionary of any keyword args for the
-            Metric classes
+        gt_data (traccuracy.TrackingGraph): ground truth graph and optionally segmentation
+        pred_data (traccuracy.TrackingGraph): predicted graph and optionally segmentation
+        matcher (Matcher): instantiated matcher object
+        metrics (List[Metric]): list of instantiated metrics objects to compute
 
     Returns:
         Dict: dictionary of metrics indexed by metric name. Dictionary will be
-        nested for metrics that return multiple values.
+            nested for metrics that return multiple values.
     """
-    if matcher_kwargs is None:
-        matcher_kwargs = {}
-    matched = matcher(**matcher_kwargs).compute_mapping(gt_data, pred_data)
-    validate_matched_data(matched, metrics)
-    metric_kwarg_dict = {
-        m_class: get_relevant_kwargs(m_class, metrics_kwargs) for m_class in metrics
-    }
+    matched = matcher.compute_mapping(gt_data, pred_data)
     results = {}
     for _metric in metrics:
-        relevant_kwargs = metric_kwarg_dict[_metric]
-        result = _metric(**relevant_kwargs).compute(matched)
-        results[_metric.__name__] = result
+        result = _metric.compute(matched)
+        results[_metric.__class__.__name__] = result
```

Comment:
What if someone runs with the same metric but different hyperparameters? That will probably happen with the new forumlation, even though it was impossible with the old one. Do we want to support this or not?

Reply:
This is a good catch. I think we should support it but I'll have to think about the right naming conventions to make it work.

Reply:
Perhaps a str function on the metric classes that does something nice?

Reply:
And/or the Metrics object optionally takes a name input?

Reply:
Let me know what you think of what I implemented. It should generate an output like this:
```
[
{'MetricName': results,
'parameters': {kwargs}},
{'MetricName': results,
'parameters': {kwargs2}}
]
```

Reply:
How about: 
```
[
{"results": results dict
"metric": metric dict including "name" entry},
]
```
This is temporary until we implement #120 but is closer to the desired access pattern.